src/coreclr/vm/methodtablebuilder.cpp

```diff
@@ -10324,16 +10321,39 @@ MethodTable * MethodTableBuilder::AllocateNewMT(
 
     MethodTable* pMT = (MethodTable*)(pData + dwGCSize);
 
-    pMT->SetMultipurposeSlotsMask(dwMultipurposeSlotsMask);
+    if (dwNumDicts != 0)
+    {
+        pMT->SetFlag(MethodTable::enum_flag_HasPerInstInfo);
+    }
+
+    if (bmtVT->pDispatchMapBuilder->Count() > 0)
+        pMT->SetFlag(MethodTable::enum_flag_HasDispatchMapSlot);
 
-    MethodTableWriteableData * pMTWriteableData = (MethodTableWriteableData *) (BYTE *)
-        pamTracker->Track(pAllocator->GetHighFrequencyHeap()->AllocMem(S_SIZE_T(sizeof(MethodTableWriteableData))));
-    // Note: Memory allocated on loader heap is zero filled
-    pMT->SetWriteableData(pMTWriteableData);
+    pMT->AllocateAuxiliaryData(pAllocator, pLoaderModule, pamTracker, fHasGenericsStaticsInfo, static_cast<WORD>(dwNonVirtualSlots), dispatchMapAllocationSize);
 
     // This also disables IBC logging until the type is sufficiently initialized so
     // it needs to be done early
-    pMTWriteableData->SetIsNotFullyLoadedForBuildMethodTable();
+    pMT->GetAuxiliaryDataForWrite()->SetIsNotFullyLoadedForBuildMethodTable();
+
+    if (bmtVT->pDispatchMapBuilder->Count() > 0)
+    {
+        DispatchMap                 *pDispatchMap        = NULL;
+        BYTE* pAllocatedSpaceAfterMethodTableAuxiliaryData = (BYTE *)(pMT->GetAuxiliaryDataForWrite() + 1);
+        // Use placement new
+        pDispatchMap = new (pAllocatedSpaceAfterMethodTableAuxiliaryData) DispatchMap(pbDispatchMapTemp, cbDispatchMapTemp);
+
+#ifdef LOGGING
+        if (dispatchMapAllocationSize.IsOverflow())
+            ThrowHR(E_FAIL);
```

Comment:
This seems odd to be behind the `LOGGING` define and it is a bit far away from where `dispatchMapAllocationSize` is defined. I would expect this up a bit.

Reply:
So, `dispatchMapAllocationSize` is a `S_SIZE_T`, and those need to be Overflow checked before they can be read. `AllocateAuxiliaryData` does that internally, but its overflow check does not escape back to this function, and so we need an additional check for overflow here. (Note: this check should never fail, as its already being used above in `AllocateAuxiliaryData`

Reply:
Of course, this doesn't even need to be an `S_SIZE_T` ...so I can fix that.